### PR TITLE
Feature/onl 09 page objects

### PR DESCRIPTION
**Pull request by AnastasiyaSoroka**
Updated Settings.java
Added BasePage
Added CartPage
Added CheckOutOverViewPage
Added CheckoutPage
Added LoginPage
Added ProductPage
Added CartTest
Added CheckoutOverviewTest
Added CheckoutTest
Updated LoginPage
Added LoginTest

### CODE REVIEW

File: src/test/java/pages/CartPage.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    private final By CHECKOUT_BUTTON = By.linkText("CHECKOUT");
+    private final By CONTINUE_BUTTON = By.className("btn_secondary");
+
+
+    public CartPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void openPage() {
+        driver.get(URL+endpoint);
```
Comment (dzmitryrak) [inline src/test/java/pages/CartPage.java:24]: URL + endpoint

File: src/test/java/pages/CartPage.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+public class CartPage extends BasePage {
+
+    private String endpoint = "cart.html";
+
+    private String priceLocator = "//*[contains(text(),'%s')]/ancestor::*[@class='cart_item']" +
+            "//div[@class='inventory_item_price']";
+    private String quantityLocator = "//*[contains(text(),'%s')]/ancestor::*[@class='cart_item']" +
+            "//div[@class='cart_quantity']";
+
+    private final By CHECKOUT_BUTTON = By.linkText("CHECKOUT");
```
Comment (dzmitryrak) [inline src/test/java/pages/CartPage.java:15]: .checkout_button

File: src/test/java/pages/CartPage.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    private final By CONTINUE_BUTTON = By.className("btn_secondary");
```
Comment (dzmitryrak) [inline src/test/java/pages/CartPage.java:16]: по тексту поиск

File: src/test/java/pages/CheckoutPage.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class CheckoutPage extends BasePage {
+
+    private final By CONTINUE_BUTTON = By.xpath("//input[@value='CONTINUE']");
```
Comment (dzmitryrak) [inline src/test/java/pages/CheckoutPage.java:8]: By.cssSelector("[value=CONTINUE]")

File: src/test/java/pages/CheckoutPage.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    private final By ERROR = By.xpath("//*[contains(text(),'Error')]");
```
Comment (dzmitryrak) [inline src/test/java/pages/CheckoutPage.java:9]: By.cssSelector("[data-test=error]")

File: src/test/java/pages/LoginPage.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class LoginPage extends BasePage {
+
+    private final By USERNAME_INPUT = By.id("user-name");
+    private final By PASSWORD_INPUT = By.id("password");
+    private final By LOGIN_BUTTON = By.id("login-button");
+    private final By ERROR = By.xpath("//*[contains(text(),'Epic')]");
```
Comment (dzmitryrak) [inline src/test/java/pages/LoginPage.java:11]: сменить на css локатор

File: src/test/java/tests/CheckoutTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
+package tests;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class CheckoutTest extends Settings {
```
Comment (dzmitryrak) [inline src/test/java/tests/CheckoutTest.java:6]: BaseTest

File: src/test/java/pages/CartPage.java (hunk 1 above)
Comment (AnastasiyaSoroka) [reply src/test/java/pages/CartPage.java:24]: Done

File: src/test/java/pages/CartPage.java (hunk 2 above)
Comment (AnastasiyaSoroka) [reply src/test/java/pages/CartPage.java:15]: done

File: src/test/java/pages/CheckoutPage.java (hunk 4 above)
Comment (AnastasiyaSoroka) [reply src/test/java/pages/CheckoutPage.java:8]: Done

File: src/test/java/pages/CheckoutPage.java (hunk 5 above)
Comment (AnastasiyaSoroka) [reply src/test/java/pages/CheckoutPage.java:9]: Done

File: src/test/java/pages/LoginPage.java (hunk 6 above)
Comment (AnastasiyaSoroka) [reply src/test/java/pages/LoginPage.java:11]: Done

File: src/test/java/tests/CheckoutTest.java (hunk 7 above)
Comment (AnastasiyaSoroka) [reply src/test/java/tests/CheckoutTest.java:6]: Done

**Pull request merged**